docs/architecture-decision-records/2023-03-integration-with-station.md

```diff
@@ -0,0 +1,260 @@
+  [Filecoin Lotus docs](https://lotus.filecoin.io/lotus/manage/manage-fil/#public-key-address) with
+  the leading `"f"` replaced with `"t"`.
+
+- Activity logs are printed to stdout with human-readable formatting.
+  ```
+  [10:30:20.000 INFO ] Connecting to the network.
+  [10:30:21.000 ERROR] Cannot connect to the orchestrator.
+  ```
+- Job completions are printed to stdout but less frequently, e.g. every 500ms.
+
+  ```
+  [10:30:20.000 STATS] Jobs completed: 123
+  [10:30:20.500 STATS] Jobs completed: 134
+  [10:30:21.000 STATS] Jobs completed: 146
+  ```
+
+## Options Considered
+
+1. Don't build a new binary, bundle the existing `zinnia` CLI inside the Station.
+
+   Pros:
+
+   - Less work to ship the first version. Less yak-shaving like setting up CI/CD workflows.
+
+   Cons:
+
+   - Zinnia modules cannot share libp2p & IPFS resources (network connection, peer address book,
+     block store).
+
+   - To meet the requirements of both module builders and the Station runtime, we would need to
+     implement extra configuration options to get different behaviour in different settings.
+
+   - Most of the Station work will be discarded later, once we need `zinniad` to allow running
+     multiple modules inside the same runtime process.
+
+2. Don't push job stats via stdout, let the station pull the stats via HTTP API. (The current
+   saturn-l2 module uses this model.)
+
+   Pros:
+
+   - We already have code in Station to deal with this.
+   - Less cluttered `stdout`
+
+   Cons:
+
+   - More complex implementation in Zinnia for little benefits. Since `zinniad` routes all
+     `console.log` messages to `stderr`, no humans should be reading `stdout`, therefore extra
+     clutter does not matter.
+
+   - More complex integration between Station and Zinnia: Zinnia needs to report URL where the stats
+     API is available, the Station needs to parse that URL from Zinnia's `stdout`.
+
+3. Include a timestamp in the JSON messages printed to `stdout` for the Station. We decided this is
+   not needed now and can be easily added later if such a need arises.
+
+   Our current goal is integration with the Station. The Station (Core or Desktop) and the Zinnia
+   runtime will initially sit on the same machine. If the log consumer sits on the same machine, it
+   shouldn't matter who will attach the timestamp.
+
+4. Use `XDG_STATE_HOME` to configure where should `zinniad` keep the state files.
+
+   The major difference between `XDG_STATE_HOME` and `STATE_ROOT` is that `XDG_STATE_HOME` provides
+   a system (or user) wide directory, we need to append a zinnia-specific segment to that path to
+   obtain `STATE_ROOT`.
+
+   I prefer to give the user full control over the location by providing them `STATE_ROOT` config
+   option.
+
+   This becomes relevant when Zinnia is running inside the Station. If we use `XDG_STATE_HOME`, then
+   we will keep the state in `XDG_STATE_HOME/zinnia`, a different place from where Station keeps its
```

Comment:
I would like to question this (above), why is it important that zinnia stores its files in the same folder as Station?

Reply:
See https://github.com/filecoin-station/zinnia/pull/121#discussion_r1143265375


> As a user running the Station on my desktop, when inspecting files created by various apps, I prefer apps to have all files in a single directory with the app name (e.g. `$HOME/Library/Caches/Filecoin\ Station`). If we use different locations for different parts of the Station, then it's not clear which directories like `zinnia` and `saturn-l2` belong to the Station and which are created by other apps.